src/Modules/OneTapLogin.php

```diff
@@ -0,0 +1,215 @@
+<?php
+/**
+ * One Tap Login Class.
+ *
+ * This class will be responsible for handling
+ * Google's one tap login for web functioning.
+ *
+ * @package RtCamp\GoogleLogin\Modules
+ * @since 1.0.16
+ */
+
+declare(strict_types=1);
+
+namespace RtCamp\GoogleLogin\Modules;
+
+use Exception;
+use RtCamp\GoogleLogin\Utils\Authenticator;
+use RtCamp\GoogleLogin\Utils\GoogleClient;
+use RtCamp\GoogleLogin\Utils\Helper;
+use RtCamp\GoogleLogin\Interfaces\Module;
+use RtCamp\GoogleLogin\Utils\TokenVerifier;
+use function RtCamp\GoogleLogin\plugin;
+
+/**
+ * Class OneTapLogin
+ *
+ * @package RtCamp\GoogleLogin\Modules
+ */
+class OneTapLogin implements Module {
+	/**
+	 * Settings Module.
+	 *
+	 * @var Settings
+	 */
+	private $settings;
+
+	/**
+	 * Token verifier.
+	 *
+	 * @var TokenVerifier
+	 */
+	private $token_verifier;
+
+	/**
+	 * Google client instance.
+	 *
+	 * @var GoogleClient
+	 */
+	private $google_client;
+
+	/**
+	 * Authenticator service.
+	 *
+	 * @var Authenticator
+	 */
+	private $authenticator;
+
+	/**
+	 * OneTapLogin constructor.
+	 *
+	 * @param Settings      $settings Settings object.
+	 * @param TokenVerifier $verifier Token verifier object.
+	 * @param GoogleClient  $client   Google client instance.
+	 * @param Authenticator $authenticator Authenticator service instance.
+	 */
+	public function __construct( Settings $settings, TokenVerifier $verifier, GoogleClient $client, Authenticator $authenticator ) {
+		$this->settings       = $settings;
+		$this->token_verifier = $verifier;
+		$this->google_client  = $client;
+		$this->authenticator  = $authenticator;
+	}
+
+	/**
+	 * Module name.
+	 *
+	 * @return string
+	 */
+	public function name(): string {
+		return 'one_tap_login';
+	}
+
+	/**
+	 * Module Initialization activity.
+	 *
+	 * Everything will happen if and only if one tap is active in settings.
+	 */
+	public function init(): void {
+		if ( $this->settings->one_tap_login ) {
+			add_action( 'login_enqueue_scripts', [ $this, 'one_tap_scripts' ] );
+			add_action( 'login_footer', [ $this, 'one_tap_prompt' ] );
+			add_action( 'wp_ajax_nopriv_validate_id_token', [ $this, 'validate_token' ] );
+			add_action( 'rtcamp.id_token_verified', [ $this, 'authenticate' ] );
+			add_action(
+				'init',
+				function () {
+					if ( ! is_user_logged_in() ) {
+						add_action( 'wp_enqueue_scripts', [ $this, 'one_tap_scripts' ] );
+						add_action( 'wp_footer', [ $this, 'one_tap_prompt' ], 10000 );
```

Comment:
Why do we need `10000` priority? Can't we keep default `10`?

Reply:
We intend it to be at very last bottom of the, so higher priority. 10 is default priority and there can be number of things in between.

Reply:
> We intend it to be at very last bottom of the, so higher priority.

Okay, but what's the other reason? Will it not work if we use default priority? IMO, we should use higher priorities only if absolutely required.